component/src/main/java/org/wso2/extension/siddhi/io/websocket/sink/websocketserver/WebSocketServerSinkConnectorListener.java

```diff
@@ -47,42 +44,36 @@
+    private List<WebSocketConnection> webSocketConnectionList = new CopyOnWriteArrayList<>();
     private String[] subProtocol = null;
     private int idleTimeout;
-    private StreamDefinition streamDefinition;
 
     WebSocketServerSinkConnectorListener(String[] subProtocol, int idleTimeout, StreamDefinition streamDefinition) {
         this.subProtocol = subProtocol;
         this.idleTimeout = idleTimeout;
-        this.streamDefinition = streamDefinition;
     }
 
     @Override
-    public void onMessage(WebSocketInitMessage initMessage) {
-        WebSocketServerHandshakeListener serverHandshakeListener = new WebSocketServerHandshakeListener(sessionList);
-        HandshakeFuture future = initMessage.handshake(subProtocol, true, idleTimeout);
-        future.setHandshakeListener(serverHandshakeListener);
+    public void onHandshake(WebSocketHandshaker webSocketHandshaker) {
+        WebSocketServerHandshakeListener serverHandshakeListener =
+                new WebSocketServerHandshakeListener(webSocketConnectionList);
+        ServerHandshakeFuture handshake = webSocketHandshaker.handshake(subProtocol, idleTimeout);
+        handshake.setHandshakeListener(serverHandshakeListener);
     }
 
     void send(Object message) {
-        sessionList.forEach(
-                currentSession -> {
-                    if (currentSession.isOpen()) {
-                        try {
-                            if (message instanceof ByteBuffer) {
-                                byte[] byteMessage = ((ByteBuffer) message).array();
-                                ByteBuffer binaryMessage = ByteBuffer.wrap(byteMessage);
-                                currentSession.getBasicRemote().sendBinary(binaryMessage);
-                            } else {
-                                currentSession.getBasicRemote().sendText((String) message);
-                            }
-                        } catch (IOException e) {
-                            throw new SiddhiAppRuntimeException("Error while sending the events defined in " +
-                                                                        streamDefinition + ".", e);
+        webSocketConnectionList.forEach(
+                currentWebSocketConnection -> {
+                    if (currentWebSocketConnection.isOpen()) {
+                        if (message instanceof ByteBuffer) {
+                            byte[] byteMessage = ((ByteBuffer) message).array();
+                            ByteBuffer binaryMessage = ByteBuffer.wrap(byteMessage);
+                            currentWebSocketConnection.pushBinary(binaryMessage);
+                        } else {
+                            currentWebSocketConnection.pushText((String) message);
                         }
                     } else {
-                        sessionList.remove(currentSession);
+                        webSocketConnectionList.remove(currentWebSocketConnection);
```

Comment:
I think this is more suitable if done in the `onClose` and `onError` resources.

Reply:
Fixed in b41fbd5